device/thunder_ripple_sdk/src/bootstrap/setup_thunder_pool_step.rs

```diff
@@ -45,18 +47,48 @@ impl ThunderPoolStep {
             warn!("Pool size of 1 is not recommended, there will be no dedicated connection for Controller events");
             return Err(RippleError::BootstrapError);
         }
-        let controller_pool =
-            ThunderClientPool::start(url.clone(), None, thunder_connection_state.clone(), 1).await;
+        let controller_pool = ripple_sdk::tokio::time::timeout(
+            Duration::from_secs(10),
+            ThunderClientPool::start(url.clone(), None, thunder_connection_state.clone(), 1),
+        )
+        .await;
+
         if controller_pool.is_err() {
```

Comment:
Do we need to the handle `Ok(Err(_)) `case here ?

Reply:
isn't that the next if block?
```
        let controller_pool = controller_pool.unwrap();
        if let Err(e) = controller_pool
```